python/dgl/sparse/sparse_matrix.py

```diff
@@ -586,6 +586,93 @@ def range_select(self, dim: int, index: slice):
             )
         raise TypeError(f"{type(index).__name__} is unsupported input type.")
 
+    def sample(
+        self,
+        dim: int,
+        n_pick: int,
+        ids: Optional[torch.Tensor] = None,
+        replacement: Optional[bool] = False,
+        bias: Optional[bool] = False,
+    ):
+        """Returns a sampled matrix on the given dimension and sample arguments.
+
+        Parameters
+        ----------
+        dim : int
+            The dimension for sampling, should be 0 or 1. `dim = 0` for
+            rowwise selection and `dim = 1` for columnwise selection.
+        n_pick : int
+            The number of elements to randomly sample on each row or column.
+        ids : torch.Tensor, optional
+            An optional tensor containing row or column IDs from which to
+            sample elements.
+            NOTE: If `ids` is not provided (i.e., `ids = None`), the function
+            will sample from all rows or columns.
+        replacement : bool, optional
+            Indicates whether repeated sampling of the same element is allowed.
+            When `replacement = True`, repeated sampling is permitted; when
+            `replacement = False`, it is not allowed.
+            NOTE: If `replacement = False` and there are fewer elements than
+            `n_pick`, all non-zero elements will be sampled.
+        bias : bool, optional
```

Comment:
Is it comment to use the value as weight or should the weight be passed in as a paramater?

Reply:
Passing weight as a parameter is also an option. But I think using the value as bias is better because
* it avoids use to think about the logic of `val_like` in `sample`. If users have a `bias` tensor, they can construct a sparse matrix by `val_like` for sampling.
* it is more clean if users want to sample a sparse matrix by its value. Otherwise, it may looks like `A.sample(..., bias=A.val)`

Reply:
SGMT.